private/pkg/storage/storagemem/storagemem.go

```diff
@@ -25,15 +25,41 @@ import (
 
 var errDuplicatePath = errors.New("duplicate path")
 
+type options struct {
+	pathToData map[string][]byte
+}
+
+// Option is provided by NewReadWriteBucket2 options.
+type Option interface {
+	apply(*options)
+}
+
+type pathData map[string][]byte
+
+func (pd pathData) apply(opts *options) {
+	opts.pathToData = pd
+}
+
+// WithFiles adds files by path to their content into the bucket.
+func WithFiles(pathToData map[string][]byte) Option {
+	return (pathData)(pathToData)
+}
+
 // NewReadWriteBucket returns a new in-memory ReadWriteBucket.
 func NewReadWriteBucket() storage.ReadWriteBucket {
 	return newBucket(nil)
 }
 
-// NewReadBucket returns a new ReadBucket.
-func NewReadBucket(pathToData map[string][]byte) (storage.ReadBucket, error) {
-	pathToImmutableObject := make(map[string]*internal.ImmutableObject, len(pathToData))
-	for path, data := range pathToData {
+// NewReadWriteBucket2 returns a new in-memory ReadWriteBucket. Errors are
+// returned with invalid options.
+func NewReadWriteBucket2(opts ...Option) (storage.ReadWriteBucket, error) {
```

Comment:
Weird constructor name, may be `NewReadWriteBucketWithOptions(WithFiles(...), )` a bit of stutter but clearer? and mark as deprecated the other one?

Reply:
I'm fine with a different suffix. The number pattern is pretty common with C and other languages which don't have overloading but you're trying to describe n ways of construction. Just in case y'all run into this in some other library. :)